### PR TITLE
endpoint: move `TestReadEpsFromDirNames` to `pkg/endpoint`

### PR DESCRIPTION
**Pull request by ianvernon**
The test was previously in \`daemon\`, and necessitated leaking a lot of
Endpoint-related state. Instead, refactor the test into the Endpoint package.
Do not regenerate the endpoints needed for testing; instead, write their
headerfiles using \`writeHeaderfile\`, since that is the operation which
is needed to be performed in order to perform the test to read the serialized
Endpoints from their corresponding headerfiles.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8997)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: pkg/endpoint/endpoint_status_test.go [hunk 1]
```diff
@@ -31,7 +31,6 @@ import (
 	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/policy"
 	"github.com/cilium/cilium/pkg/policy/trafficdirection"
-
```
Comment (raybejjani) [inline pkg/endpoint/endpoint_status_test.go]: This is unintended, I think?

**Review by raybejjani — APPROVED** [review]

File: pkg/endpoint/endpoint_status_test.go (hunk 1 above)
Comment (ianvernon) [reply pkg/endpoint/endpoint_status_test.go]: Fixed!

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**